src/components/TodoFooter.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import { FilterStatus } from '../types/FilterStatus';
+import classNames from 'classnames';
+
+type Props = {
+  filterStatus: FilterStatus;
+  setFilterStatus: Dispatch<SetStateAction<FilterStatus>>;
+  todosLeft: number;
+};
+
+export const TodoFooter: React.FC<Props> = props => {
+  const { filterStatus, setFilterStatus, todosLeft } = props;
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {todosLeft} items left
+      </span>
+
+      {/* Active link should have the 'selected' class */}
+      <nav className="filter" data-cy="Filter">
+        {Object.values(FilterStatus).map(filter => (
+          <a
+            key={filter}
+            href={`#/${filter === FilterStatus.All ? '' : filter.toLocaleLowerCase()}`}
+            className={classNames('filter__link', {
+              selected: filterStatus === filter,
+            })}
+            data-cy={`FilterLink${filter}`}
+            onClick={() => setFilterStatus(filter)}
+          >
+            {filter}
+          </a>
+        ))}
+      </nav>
+
+      {/* this button should be disabled if there are no completed todos */}
+      <button
+        type="button"
+        className="todoapp__clear-completed"
+        data-cy="ClearCompletedButton"
+      >
+        Clear completed
+      </button>
```

Comment:
The 'Clear completed' button should be disabled if there are no completed todos, as per the task checklist. Consider adding a `disabled` attribute to the button, which is conditionally set based on whether there are any completed todos.